native_field/src/lib.rs

```diff
@@ -0,0 +1,266 @@
+#![no_std]
+
+extern crate alloc;
+
+use alloc::vec;
+use alloc::vec::Vec;
+use columns::{NativeFieldCols, COL_MAP, NUM_COLS};
+use core::mem::transmute;
+use valida_bus::{MachineWithGeneralBus, MachineWithRangeBus8};
+use valida_cpu::MachineWithCpuChip;
+use valida_machine::{instructions, Chip, Instruction, Interaction, Machine, Operands, Word};
+use valida_range::MachineWithRangeChip;
+
+use p3_air::VirtualPairCol;
+use p3_field::{Field, PrimeField32};
+use p3_matrix::dense::RowMajorMatrix;
+use p3_maybe_rayon::*;
+
+const ADD_OPCODE: u32 = 13;
+const SUB_OPCODE: u32 = 14;
+const MUL_OPCODE: u32 = 15;
+
+pub mod columns;
+pub mod stark;
+
+#[derive(Clone)]
+pub enum Operation {
+    Add(Word<u8>, Word<u8>, Word<u8>),
+    Sub(Word<u8>, Word<u8>, Word<u8>),
+    Mul(Word<u8>, Word<u8>, Word<u8>),
+}
+
+pub struct NativeFieldChip {
+    operations: Vec<Operation>,
+}
+
+impl<F, M> Chip<M> for NativeFieldChip
+where
+    F: Field,
+    M: MachineWithGeneralBus<F = F> + MachineWithRangeBus8,
+{
+    fn generate_trace(&self, _machine: &M) -> RowMajorMatrix<M::F> {
+        let rows = self
+            .operations
+            .par_iter()
+            .map(|op| self.op_to_row(op))
```

Comment:
Maybe better to `flat_map(...).flatten().collect()` to avoid the intermediate `collect()`? But also feel free to leave for now, we can optimize later after adding benchmarks

Reply:
Will hold off for now until we have benchmarks. I remember some odd behavior with rayon, where flattening before collecting had worse performance (~25% slower on my machine IIRC). Making all other changes you noted below. Thanks.